libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Identity/SFIdentityData.m

```diff
@@ -322,20 +321,19 @@ - (NSDate *)lastModifiedDate
 
 - (NSURL *)parentExistsOrNilForUrl:(NSString *)parentKey childKey:(NSString *)childKey
 {
-    NSDictionary *parentDict = (self.dictRepresentation)[parentKey];
-    if (parentDict != nil)
-        return [NSURL URLWithString:parentDict[childKey]];
-    else
-        return nil;
+    NSString *value = [self parentExistsOrNilForString:parentKey childKey:childKey];
+    return value != nil ? [NSURL URLWithString:value] : nil;
 }
 
 - (NSString *)parentExistsOrNilForString:(NSString *)parentKey childKey:(NSString *)childKey
 {
-    NSDictionary *parentDict = (self.dictRepresentation)[parentKey];
-    if (parentDict != nil)
-        return parentDict[childKey];
-    else
+    NSDictionary *parentDict = [self.dictRepresentation nonNullObjectForKey:parentKey];
+    if (parentDict != nil) {
+        NSString *value = [parentDict nonNullObjectForKey:childKey];
+        return [value isKindOfClass:[NSString class]] ? value : nil;
```

Comment:
we only need to do return [parentDict nonNullObjectForKey:chilKey] (that will take care of returning nil if it is a NSNull).

Reply:
`nonNullObjectForKey` will not take care of the case where object is `NSNumber` or `NSDictionary` or anything else that's not `NSnull`. Also, given that `parentExistsOrNilFor**String**` is a helper method specifically guaranteeing that result is either `nil` or `NSString`, the extra check here makes sense to me.

Reply:
Ok.